### PR TITLE
wedge buffers support start and end angles

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

File: src/core/geometry/qgsgeometry.h [hunk 1]
```diff
@@ -345,6 +345,22 @@ class CORE_EXPORT QgsGeometry
...
+     *
+     * The outer radius of the buffer is specified via \a outerRadius, and optionally an
+     * \a innerRadius can also be specified.
+     *
+     * The returned geometry will be a CurvePolygon geometry containing circular strings. It may
+     * need to be segmentized to convert to a standard Polygon geometry.
+     *
+     * \since QGIS 3.40
+     */
+    static QgsGeometry createWedgeBufferV2( const QgsPoint &center, double startAngle, double endAngle,
```
Comment (nyalldawson) [inline src/core/geometry/qgsgeometry.h:361]: I don't think this should be V2, because it doesn't replace the other. Maybe createWedgeBufferFromAngles?

**Review by 3nids — COMMENTED** [review]
support for geographic coordinates

File: src/core/geometry/qgsgeometry.h [hunk 2]
```diff
@@ -30,6 +30,7 @@ email                : morb at ozemail dot com dot au
 
 #include "qgis_core.h"
 #include "qgis_sip.h"
+#include "qgscoordinatereferencesystem.h"
```
Comment (3nids) [inline src/core/geometry/qgsgeometry.h:33]: is this OK to add this include? it seems QgsGeometry was pretty light in terms of includes?

Comment (nyalldawson) [reply src/core/geometry/qgsgeometry.h:33]: I'd prefer not, but for different reasons 😁. There's a big warning at the top of this class that everything is Cartesian only, and I'm concerned if we start mixing in non Cartesian methods then the situation will get very murky. I'd lean toward putting this new function in a different utility class dedicated for ellipsoid aware geometry utils.

Comment (3nids) [reply src/core/geometry/qgsgeometry.h]: I've dropped the ellipsoïd part.

File: src/core/geometry/qgsgeometry.cpp [hunk 3]
```diff
@@ -409,6 +409,26 @@ QgsGeometry QgsGeometry::collectGeometry( const QVector< QgsGeometry > &geometri
...
+  startAngle = QgsGeometryUtilsBase::normalizedAngle( startAngle * M_PI / 180 ) * 180 / M_PI;
+  endAngle = QgsGeometryUtilsBase::normalizedAngle( endAngle * M_PI / 180 ) * 180 / M_PI;
+
+  double angularWidth = endAngle - startAngle;
+  double averageAngle = QgsGeometryUtilsBase::averageAngle( endAngle * M_PI / 180, startAngle * M_PI / 180 ) * 180 / M_PI;
+
+  bool useShortestArc = ( angularWidth >= 0 && angularWidth <= 180.0 ) || ( angularWidth <= -180.0 && angularWidth >= -360.0 );
+  if ( !useShortestArc )
+    averageAngle += 180;
+
```
Comment (lbartoletti) [inline src/core/geometry/qgsgeometry.cpp:431]: You could consitfy, and use const variables here

```suggestion
```suggestion
  const double HALF_CIRCLE_DEGREES = 180.0;
  const double FULL_CIRCLE_DEGREES = 360.0;
  // maybe we have/need a method for these two:
  const double DEG_TO_RAD = M_PI / HALF_CIRCLE_DEGREES;
  const double RAD_TO_DEG = HALF_CIRCLE_DEGREES / M_PI;

  startAngle = QgsGeometryUtilsBase::normalizedAngle(startAngle * DEG_TO_RAD) * RAD_TO_DEG;
  endAngle = QgsGeometryUtilsBase::normalizedAngle(endAngle * DEG_TO_RAD) * RAD_TO_DEG;

  const double angularWidth = endAngle - startAngle;
  const bool useShortestArc = (angularWidth >= 0 && angularWidth <= HALF_CIRCLE_DEGREES) || 
                            (angularWidth <= -HALF_CIRCLE_DEGREES && angularWidth >= -FULL_CIRCLE_DEGREES);

  const double averageAngle = QgsGeometryUtilsBase::averageAngle(endAngle * DEG_TO_RAD, startAngle * DEG_TO_RAD) * RAD_TO_DEG +
                            (useShortestArc ? 0 : HALF_CIRCLE_DEGREES);

```

**Review by nirvn — APPROVED** [review]
Looking good -- let's make sure this is squashed :wink:

**Pull request merged**